x-pack/plugins/data_visualizer/public/application/index_data_visualizer/hooks/use_overall_stats.ts

```diff
@@ -116,72 +142,63 @@ export function useOverallStats<TParams extends OverallStatsSearchStrategyParams
       abortSignal: abortCtrl.current.signal,
       sessionId: searchStrategyParams?.sessionId,
     };
-    const nonAggregatableOverallStats$ =
-      nonAggregatableFields.length > 0
-        ? forkJoin(
-            nonAggregatableFields.map((fieldName: string) =>
-              data.search
-                .search<IKibanaSearchRequest, IKibanaSearchResponse>(
-                  {
-                    params: checkNonAggregatableFieldExistsRequest(
-                      index,
-                      searchQuery,
-                      fieldName,
-                      timeFieldName,
-                      earliest,
-                      latest,
-                      runtimeFieldMap
-                    ),
-                  },
-                  searchOptions
-                )
-                .pipe(
-                  switchMap((resp) => {
-                    return of({
-                      ...resp,
-                      rawResponse: { ...resp.rawResponse, fieldName },
-                    } as IKibanaSearchResponse);
-                  })
-                )
-            )
-          )
-        : of(undefined);
+
+    const nonAggregatableFieldsObs = nonAggregatableFields.map((fieldName: string) =>
```

Comment:
There seems to be an issue with loading data for the expanded rows for non aggregatable fields, where only a limited number of them get populated. Here on the Kibana ecommerce sample data set, I only see examples for five `text` fields:

![image](https://user-images.githubusercontent.com/7405507/158863554-c75689d8-a4b3-4afd-aff1-9142bb095de2.png)

Reply:
Thanks for catching that. Fixed here https://github.com/elastic/kibana/pull/124898/commits/38de83e47023954436ef12513360476538c0d205